calcom/cal.com
# feat: min max chars for long text

/claim https://github.com/calcom/cal.com/issues/15577

## What does this PR do?

This PR adds an optional minimum and maximum character limit to long text question in booking form.



- Fixes #15577
- Fixes [CAL-3973](https://linear.app/calcom/issue/CAL-3973/booking-questions-for-long-text-add-min-and-max-characters-both)



Loom Video: https://www.loom.com/share/735a6768a92847cabefd1fe2187c499d?sid=2d606562-c6e6-4bd3-92c0-b1fcdab14218


## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- Are there environment variables that should be set?
No

- What are the minimal test data to have?
N.A.

- What is expected (happy path) to have (input and output)?
1. Click on `Event Types` in the sidebar.
2. Click on an event.
3. Inside the event, click on the `Advanced` tab on the left sidebar.
4. Add / edit a question.
5. Set type to `Long text`
6. Set the min &/or max characters.
7. Save the question.
8. Save the form.
9. Click on the `Preview` button in the event header.
10. Select a date on the calendar.
11. While filling out the form, click save button with:
a. Min &/or max character limit unsatisfied -> should show red text with error.
b. All the conditions satisfied -> Should save successfully.

## Checklist
